templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -482,31 +491,42 @@ contents:
       # Some deployments uses a temporary solution where br-ex is moved out from the default gateway interface
       # and bound to a different nic. If that is the case, we rollback (https://github.com/trozet/openshift-ovn-migration).
       if [ "$iface" != "br-ex" ] && [  -f "$extra_bridge_file" ] && nmcli connection show br-ex &> /dev/null; then
-        echo "default gateway is not bridge but bridge exists, reverting"
-        update_nm_conn_files br-ex phys0
-        rm_nm_conn_files
-        nmcli c reload
-        sleep 5
+        echo "Default gateway is not bridge but bridge exists, reverting"
+        rollback
+        reload_nm
+        print_state
       fi
 
       convert_to_bridge "$iface" "br-ex" "phys0"
+
+      # Check if we need to configure the second bridge
       if [ -f "$extra_bridge_file" ] && (! nmcli connection show br-ex1 &> /dev/null || ! nmcli connection show ovs-if-phys1 &> /dev/null); then
         interface=$(head -n 1 $extra_bridge_file)
         convert_to_bridge "$interface" "br-ex1" "phys1"
       fi
-      if [ ! -f "$extra_bridge_file" ] && (nmcli connection show br-ex1 &> /dev/null || nmcli connection show ovs-if-phys1 &> /dev/null); then # the file was removed
+
+      # Check if we need to remove the second bridge
+      if [ ! -f "$extra_bridge_file" ] && (nmcli connection show br-ex1 &> /dev/null || nmcli connection show ovs-if-phys1 &> /dev/null); then
         update_nm_conn_files br-ex1 phys1
         rm_nm_conn_files
-        nmcli c reload
-        sleep 5
       fi
 
       # remove bridges created by openshift-sdn
       ovs-vsctl --timeout=30 --if-exists del-br br0
-      print_state
-      exit 0
+      
+      # recycle NM connections
+      reload_nm
+
+      # make sure everything is activated
+      activate_nm_conn ovs-if-phys0
+      activate_nm_conn ovs-if-br-ex
+      if [ -f "$extra_bridge_file" ]; then
+        activate_nm_conn ovs-if-phys1
+        activate_nm_conn ovs-if-br-ex1
+      fi
     elif [ "$1" == "OpenShiftSDN" ]; then
       # Revert changes made by /usr/local/bin/configure-ovs.sh during SDN migration.
       rollback
-      print_state
+      reload_nm
+      ovs-vsctl --timeout=30 --if-exists del-br br-int -- --if-exists del-br br-local
```

Comment:
I see you moved this down to SDN specific. Should clean it up from the first commit